### ISSUE TITLE
files predictions_ensemble_*.npy take all available disk space

### ISSUE THREAD

**Issue by Motorrat** [automl/auto-sklearn#665]
I am starting binary classification auto-sklearn in parallel mode with a somewhat larger data-set (470K rows) and  `atsklrn_tmp/.auto-sklearn/predictions_ensemble/predictions_ensemble_*.npy` files overfill all available disk space after a few hours. Then auto-sklearn fails. 
The largest disk I have tried so far is 100GB. The individual flies predictions_ensemble_*.npy are all of the same size (about 1GB) for example `1174189 Apr 20 09:49 predictions_ensemble_8_11.npy`

Here is my code, it is running auto-sklearn 0.5.1.
https://github.com/paypal/autosklearn-zeroconf/blob/master/bin/dataTransformationProcessing.py#L129

I can send over the data-set if needed. It is 18MB of size.

The error message is trivial but I am including it here for completeness.
```
2019-04-20 01:24:36,516 - [ZEROCONF] - spawn_autosklearn_classifier - ERROR - Error in clf.fit - seed:13
Traceback (most recent call last):
  File "/home/user/zeroconf/lib/python3.6/site-packages/smac/facade/smac_facade.py", line 400, in optimize
    incumbent = self.solver.run()
  File "/home/user/zeroconf/lib/python3.6/site-packages/smac/optimizer/smbo.py", line 196, in run
    output_directory=self.scenario.output_dir_for_this_run)
  File "/home/user/zeroconf/lib/python3.6/site-packages/smac/optimizer/pSMAC.py", line 83, in write
    run_history.save_json(temporary_filename, save_external=False)
  File "/home/user/zeroconf/lib/python3.6/site-packages/smac/runhistory/runhistory.py", line 333, in save_json
    "configs": configs}, fp, cls=EnumEncoder)
  File "/usr/lib/python3.6/json/__init__.py", line 180, in dump
    fp.write(chunk)
OSError: [Errno 28] No space left on device

During handling of the above exception, another exception occurred:

OSError: [Errno 28] No space left on device

During handling of the above exception, another exception occurred:

Trace
```

**Comment by Motorrat** [comment]
After changing some seemingly unrelated logic in data preparation I can't reproduce the above behaviour anymore. Currently 1M rows of essentially same data only takes about 20GB on the disk in the temp-directory and this amount seems stable.

**Comment by mfeurer** [comment]
It appears to be a duplicate of #454 which is basically caused by not deleting files which are no longer needed (and at some point the "ensemble prediction" files are no longer needed). However, the numbers you gave (470k rows, 1GB size) imply that the predictions for a single row are between 2 and 2.5 MB. Is this a multilabel classification problem? Could you please load one of the pickle files into memory and check the data type and dimensions of the array?

**Comment by Motorrat** [comment]
Yes it looks like #454. It is a binary classification problem. Unfortunately I can not reproduce it anymore. thank you for the instructions on the analysis. I will write back if the problem re-appears.

**Comment by mfeurer** [comment]
Hey, it took quite a while, but we finally added a new argument `max_models_on_disc` to Auto-sklearn, which governs the number of models and predictions kept on disc (https://automl.github.io/auto-sklearn/master/api.html). 

Please reopen if there are still issues with this.
